### ISSUE TITLE
[Bug] downloaded media files are not indexed to MediaStorage 

### ISSUE THREAD

**Issue by ChuggaChoo** [mozilla-mobile/android-components#8343]
Downloaded media files are not indexed to [MediaStorage](https://developer.android.com/reference/android/provider/MediaStore)
## Steps to reproduce

download any image from web
### Expected behavior

shows up on gallery
### Actual behavior

nothing shows up.
### Device information

* Android device: Android 10 (OnePlus 7T)

* Mozilla Daylight (80.1.2)

Note: This appears to be an Old Issue that's happening again?
https://github.com/mozilla-mobile/fenix/issues/8754

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-7216)

**Comment by fersosam** [comment]
I have this problem on at least two devices. I just noticed it on the first device about two days ago. Chrome has no problems (the only other browser installed). If I reset the device the files that had already been downloaded but not index are finally indexed. But any new download after reboot is no longer indexed again, a new reboot is required to index the latest downloads.

The first device has Android Firefox 80.1.2 (Build #2015761283) and is a Samsung Galaxy S20 Plus with One UI 2.5, Android 10, Kernel 4.19.87-19350896, and update G985FXXU4BTH5 / G985FOWA4BTH6 / G985FXXU4BTH5 (august 1, 2020).

The second device has Android Firefox 80.1.2 (Build #2015761281) and is a Samsung Galaxy Tab A (2019) with One UI 2.1, Android 10, Kernel 4.4.177-19000394, and update T510XXU3BTFN / T5100WO3BTG1 / (july 1, 2020).

Both devices are updated as much as Samsung and the Google Play store allow.

On the S20+ I downloaded Firefox Beta and had the same problem (I already uninsulated it, I don't remember exact version and build, sorry).

Hope this helps.
